pkg/operator/sync_v310_00.go

```diff
@@ -23,6 +23,100 @@ import (
+		mergedVersionAvailability.UpdatedReplicas = 1
+	}
+	if signingVersionAvailability.ReadyReplicas > 0 && apiServiceInjectorVersionAvailability.ReadyReplicas > 0 {
+		mergedVersionAvailability.ReadyReplicas = 1
+	}
+
+	return mergedVersionAvailability, allErrors
+}
+
+func sync_v310_00_to_00_APIServiceController(c ServiceCertSignerOperator, operatorConfig *scsv1alpha1.ServiceCertSignerOperatorConfig, previousAvailability *operatorsv1alpha1.VersionAvailablity) (operatorsv1alpha1.VersionAvailablity, []error) {
+	versionAvailability := operatorsv1alpha1.VersionAvailablity{
+		Version: operatorConfig.Spec.Version,
+	}
+
+	errors := []error{}
+	var err error
+
+	requiredNamespace := resourceread.ReadNamespaceV1OrDie(v310_00_assets.MustAsset("v3.10.0/apiservice-cabundle-controller/ns.yaml"))
+	_, _, err = resourceapply.ApplyNamespace(c.corev1Client, requiredNamespace)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "ns", err))
+	}
+
+	requiredService := resourceread.ReadServiceV1OrDie(v310_00_assets.MustAsset("v3.10.0/apiservice-cabundle-controller/svc.yaml"))
+	_, _, err = resourceapply.ApplyService(c.corev1Client, requiredService)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "svc", err))
+	}
+
+	requiredSA := resourceread.ReadServiceAccountV1OrDie(v310_00_assets.MustAsset("v3.10.0/apiservice-cabundle-controller/sa.yaml"))
+	_, saModified, err := resourceapply.ApplyServiceAccount(c.corev1Client, requiredSA)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "sa", err))
+	}
+
+	// TODO create a new configmap whenever the data value changes
+	_, configMapModified, err := ensureAPIServiceInjectorConfigMap_v310_00_to_00(c, operatorConfig.Spec)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "cm", err))
+	}
+
+	_, signingCABundleModified, err := manageSigningCABundle(c)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "signing-key", err))
+	}
+
+	forceDeployment := operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration
+	if saModified { // SA modification can cause new tokens
+		forceDeployment = true
+	}
+	if signingCABundleModified {
+		forceDeployment = true
+	}
+	if configMapModified {
+		forceDeployment = true
+	}
+
+	// our configmaps and secrets are in order, now it is time to create the DS
+	// TODO check basic preconditions here
+	actualDeployment, _, err := ensureAPIServiceInjectorDeployment_v310_00_to_00(c, operatorConfig, previousAvailability, forceDeployment)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "deployment", err))
+	}
+	if actualDeployment != nil {
+		versionAvailability.UpdatedReplicas = actualDeployment.Status.UpdatedReplicas
+		versionAvailability.ReadyReplicas = actualDeployment.Status.ReadyReplicas
+		versionAvailability.Generations = []operatorsv1alpha1.GenerationHistory{
+			{
+				Group: "apps", Resource: "Deployment",
```

Comment:
do we need a group version here?

Reply:
> do we need a group version here?

No.  We assume that all versions have the same conceptual view of a Deployment